pkg/reconciler/buildrun/buildrun_test.go

```diff
@@ -1216,6 +1214,271 @@ var _ = Describe("Reconcile BuildRun", func() {
 				Expect(err).ToNot(BeNil())
 				Expect(resources.IsClientStatusUpdateError(err)).To(BeTrue())
 			})
+
+			It("should mark buildrun succeeded false when BuildRun name is too long", func() {
+				buildRunSample = ctl.BuildRunWithoutSA("f"+strings.Repeat("o", 63)+"bar", buildName)
+
+				client.GetCalls(ctl.StubBuildRun(buildRunSample))
+				statusWriter.UpdateCalls(func(_ context.Context, o crc.Object, _ ...crc.UpdateOption) error {
+					Expect(o).To(BeAssignableToTypeOf(&build.BuildRun{}))
+					switch buildRun := o.(type) {
+					case *build.BuildRun:
+						condition := buildRun.Status.GetCondition(build.Succeeded)
+						Expect(condition.Reason).To(Equal(resources.BuildRunNameInvalid))
+						Expect(condition.Message).To(Equal("must be no more than 63 characters"))
+					}
+
+					return nil
+				})
+
+				_, err := reconciler.Reconcile(context.TODO(), buildRunRequest)
+				Expect(err).To(BeNil())
```

Comment:
Hm, I just notice that this is missing at many places including existing tests. But don't we need this to verify that the update call happened at all?

```suggestion
				Expect(err).To(BeNil())
				Expect(client.UpdateStatusCallCount()).To(Equal(1))
```

Reply:
I wrote the test cases in such a way that they fail with the `Expect`, so that I know the respective section is hit. But, I agree, it is A) cleaner/more precise to have the check as well and B) helps with potential regressions when common test code is touched. Will add the check